src/plugins/discover/public/application/angular/context/api/anchor.ts

```diff
@@ -6,11 +6,27 @@
  * Side Public License, v 1.
  */
 
-import _ from 'lodash';
+import _, { Dictionary } from 'lodash';
 import { i18n } from '@kbn/i18n';
 
-export function fetchAnchorProvider(indexPatterns, searchSource, useNewFieldsApi = false) {
-  return async function fetchAnchor(indexPatternId, anchorId, sort) {
+import { ISearchSource, IndexPatternsContract, SortDirection } from '../../../../../../data/public';
+import { EsHitRecord } from './context';
+
+export interface AnchorHitRecord extends EsHitRecord {
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  $$_isAnchor: boolean;
+}
+
+export function fetchAnchorProvider(
+  indexPatterns: IndexPatternsContract,
+  searchSource: ISearchSource,
+  useNewFieldsApi: boolean = false
+) {
+  return async function fetchAnchor(
+    indexPatternId: string,
+    anchorId: string,
+    sort: [Dictionary<SortDirection>, { [key: string]: SortDirection }]
```

Comment:
You could import 
```
import { EsQuerySortValue } from '../../../../../../data/public';
```
and switch to
```
sort: EsQuerySortValue[]
```
here, so Dictionary wouldn't be needed, and code would be further simplified 🥳